src/Strategy.js

```diff
@@ -0,0 +1,73 @@
+//@flow
+
+import {expandStyle} from './compiler';
+import {Spec} from './ElementPropSpec';
+import {staticStylesheetManager, dynamicStylesheetManager} from './StylesheetManager';
+
+export const resolve = ({
+  defaultClassName,
+  defaultComponent,
+  ownProps,
+}: {
+  defaultClassName: Array<string>,
+  defaultComponent: *,
+  ownProps: *,
+}) => {
+  let Component = defaultComponent;
+  const className = defaultClassName;
```

Comment:
I meant `className` array specifically here. But I agree, having `Component` specified only though `props` is superior and less confusing.

Currently it looks it up in `props` and falls back to `constructor.Component` if nothing found. I made this originally to allow easy overrides of the `Component` via subclassing:
```
class FancyButton extends Button {
  static Component = FancyButtonBaseOrSomething;
}
```
Now I think this was a bad idea.